src/biogeophys/CanopyHydrologyMod.F90

```diff
@@ -645,12 +675,12 @@ subroutine FracWet(numf, filter, canopystate_inst, waterstatebulk_inst, waterdia
 
              if (h2ocan > 0._r8) then
                 vegt    = frac_veg_nosno(p)*(elai(p) + esai(p))
-                dewmxi  = 1.0_r8/dewmx(p)
+                dewmxi  = 1.0_r8/params_inst%dewmx  ! TEMP for BFB; rm bef merge
                 fwet(p) = ((dewmxi/vegt)*h2ocan)**0.666666666666_r8
                 fwet(p) = min (fwet(p),maximum_leaf_wetted_fraction)   ! Check for maximum limit of fwet
                 if (snocan(p) > 0._r8) then
-                   dewmxi  = 1.0_r8/dewmx(p)
-                   fcansno(p) = ((dewmxi/(vegt*6.0_r8*10.0_r8))*snocan(p))**0.15_r8 ! must match snocanmx 
+                   dewmxi  = 1.0_r8/params_inst%dewmx  ! TEMP for BFB; rm bef merge
+                   fcansno(p) = ((dewmxi / (vegt * params_inst%sno_stor_max * 10.0_r8)) * snocan(p))**0.15_r8 ! must match snocanmx 
                    fcansno(p) = min (fcansno(p),1.0_r8)
```

Comment:
@billsacks and @ekluzek 
I have spotted this piece of code that seems wasteful. We could remove dewmxi and change
`fwet(p) = ((dewmxi/vegt)*h2ocan)**0.666666666666_r8`
and
`fcansno(p) = ((dewmxi / (vegt * params_inst%sno_stor_max * 10.0_r8)) * snocan(p))**0.15_r8`
to 
`fwet(p) = (h2ocan / vegt)**0.666666666666_r8`
and
`fcansno(p) = (snocan(p) / (vegt * params_inst%sno_stor_max * 10.0_r8))**0.15_r8`

Reply:
These changes would not be BFB. Should I add comments in the code for a future clean-up?

Reply:
I'm confused: your new versions don't seem to have dewmx in them at all. Is that a mistake?

I'm guessing that this is written the way it is in an attempt to avoid divisions, since division is a lot more expensive than multiplication. The way it's written now, it probably doesn't accomplish anything. But if the inverse was taken outside of the loop, then it would have some performance benefit to keep this as a multiplication by dewmxi rather than a division by dewmx.

Reply:
You're right, thank you for catching my omission. I have edited my suggested mods to now include params_inst%dewmx in the denominator.

So, yes, in fact I think it's more wasteful than necessary as written, in part because it calculates dewmxi twice in the same loop; also because it calculates dewmxi at all instead of using dewmx in a division that would happen regardless.

If we're attached to keeping dewmxi, then moving it out of the p loop would be more efficient, as you suggest.

Reply:
I think this is something we can come back to later. I may do it in some upcoming cleanup I'll be doing of this module, or we can come back to this at a later time.